dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/registry/impl/ServiceRegistryActor.scala

```diff
@@ -44,41 +44,48 @@ object InternalRegistry {
   def build(unmanagedServices: UnmanagedServices): InternalRegistry = new InternalRegistry(
     unmanagedServices.services.flatMap {
       case (serviceName, serviceRegistryService) =>
-        serviceRegistryService.uris().asScala.flatMap {
-          buildRegistryItem(serviceName, serviceRegistryService)
+        serviceRegistryService.uris().asScala.flatMap { uri =>
+          // External services are registeredtwice!
+          // Once without a portName (for lookups using ServiceLocator, eg: Lagom client)
+          // and once with a portName based on uri scheme (for lookup using ServiceDiscovery, eg: gRPC clients)
+          val portNames = Seq(None, Some(uri.getScheme))
+          buildRegistryItem(serviceName, serviceRegistryService, portNames, uri)
         }
     }
   )
 
-  private def buildRegistryItem(serviceName: String, serviceRegistryService: ServiceRegistryService)(
+  private def buildRegistryItem(
+      serviceName: String,
+      serviceRegistryService: ServiceRegistryService,
+      portNames: Seq[Option[String]],
       serviceUri: URI
   ): Seq[(ServiceRegistryKey, ServiceToRegister)] = {
-    val portNames = {
-      serviceUri.getScheme match {
-        case "tcp" =>
-          Seq(None) // using "tcp://" defaults to having no portName
-        case "http" =>
-          // using "http://" defaults to having no portName and also Some("http") so "http" becomes the
-          // default result when searching without a `portName` query.
-          Seq(None, Some(serviceUri.getScheme))
-        case _ => Seq(Some(serviceUri.getScheme))
-      }
-    }
     val registryItems: Seq[(ServiceRegistryKey, ServiceToRegister)] = portNames.map { pn =>
       val srk = ServiceRegistryKey(ServiceName(serviceName), pn)
       srk -> ServiceToRegister(serviceUri, serviceRegistryService)
     }
     registryItems
   }
 
-  def build(serviceName: String, details: ServiceRegistryService): Map[ServiceRegistryKey, ServiceToRegister] =
+  def build(serviceName: String, details: ServiceRegistryService): Map[ServiceRegistryKey, ServiceToRegister] = {
     details
       .uris()
       .asScala
-      .flatMap {
-        buildRegistryItem(serviceName, details)
+      .flatMap { uri =>
+        val portNames =
+          uri.getScheme match {
+            case "tcp" =>
+              Seq(None) // using "tcp://" defaults to having no portName
+            case "http" =>
+              // using "http://" defaults to having no portName and also Some("http") so "http" becomes the
+              // default result when searching without a `portName` query.
+              Seq(None, Some(uri.getScheme))
+            case _ => Seq(Some(uri.getScheme))
```

Comment:
This code is moved to here. This is exactly as before, but now only applying to managed services. 

However, I do think that this should be reduced to always use `None` for any managed services. The reason being that any managed service is looked up using Lagom's ServiceLocator and therefore the `portName` will always be `None. 

The case `case "http" => Seq(None, Some(uri.getScheme))` is registering it twice, but there is no way that port name `http` (eg: `Some(uri.getScheme)`) can ever be used. 

The case `case _ => Seq(Some(uri.getScheme))` is even worst, basically we are saying: "anythyng that is not tcp or http, gets a portName and therefore is invible to the ServiceLocator`. This becomes explicit on this test case: https://github.com/lagom/lagom/pull/2491/files#diff-4ecf98a69956bdf9cd38b90ee4be4a98R184-R193

There are ways to build a smart ServiceLocator with some hardcoded `portNames`, but I don't think this is something we should even support. In other to hack it out a user would need to register services themselves, by first disabling service registration in Lagom, than add their own registration process, then adding their own smart ServiceLocator. I don't think we should have code to support such a use case.

Reply:
The Long-term solution/implementation is for every service in the DevMode's Service REgistry to have a `portName` and fr that to no longer be an `Option[String]`.

The current code as it is serves both `ServiceLocator` (has no concept of `portName` on the API) and `ServiceDiscovery`.

Reply:
I agree that we should have it as long term solution, but this solution as it is now is only adding code, making it hard to understand the 'magic' and zero added value. 

That's why I'm in favour of removing it. We should not have code that brings zero value. In that specific case, it's even adding a strange behaviour. Managed services with https become invisible.  

Doesn't matter if in the future we will make it different or need to support it. The actual situation is that `portName` are simply too magical,  hacky and buggy to make it work with ServiceLocator.